### PR TITLE
std.format: Repair docs of compound indicator and add examples.

### PR DESCRIPTION
**Pull request by berni44**
Last one in a long row of rework document `std.format` PRs!

While trying to unify the descriptions of compound specifiers I realized, that I erred on the use of them. So I mainly took the former description (which was updated in PR #7944) and merged it somehow with my former description.

I moved all examples to real unittests and added several more to cover most of the stuff explained before.

Finally I added a release note.

### CODE REVIEW

File: changelog/rework_format_docs.dd [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+Some highlights:
+
+$(UL
+$(LI The grammar of the format string was updated.)
+$(LI A detailed description of format specifiers was provided.)
+$(LI Several examples on how to use the functions and the format
+     strings were added.)
+$(LI The first example(s) of each item were reworked to match the
+     needs of inexperienced programmers.)
```
Comment (RazvanN7) [inline changelog/rework_format_docs.dd:16]: I would rephrase or even drop this last item. If you want to have this in the changelog it should not contain negative attributes like "inexperienced". Something along the lines of: "The examples for each function have been reworked to provide a smooth transition from simple, yet comprehensive, use cases to more complex and difficult scenarios."

File: std/format/package.d [hunk 2]
```diff
@@ -187,16 +187,24 @@ $(BOOKTABLE ,
...
+indicate the start of the delimiter, so that the preceding portion of
+the string will be included following the last element.
+
 
 The $(I format string) inside of the $(I compound indicator) should
-contain exactly one $(I format specifier), which specifies the way,
-the elements of the compound type will be formatted. This $(I format
-specifier) can be a $(I compound indicator) itself.
+contain exactly one $(I format specifier) (two in case of associative
+arrays), which specifies the way, the elements of the compound type
```
Comment (RazvanN7) [inline std/format/package.d:201]:
```suggestion
arrays), which specifies the formatting mode of the elements of the compound type
```

File: std/format/package.d [hunk 3]
```diff
@@ -187,16 +187,24 @@ $(BOOKTABLE ,
...
+will be formatted. This $(I format specifier) can be a $(I compound
```
Comment (RazvanN7) [inline std/format/package.d:202]:
```suggestion
. This $(I format specifier) can be a $(I compound
```

File: std/format/package.d [hunk 4]
```diff
@@ -510,128 +518,105 @@ $(BOOKTABLE ,
    )
 )
 
+Copyright: Copyright The D Language Foundation 2000-2013.
```
Comment (RazvanN7) [inline std/format/package.d:521]:
```suggestion
Copyright: Copyright The D Language Foundation 2000-2021.
```

**Review by RazvanN7 — CHANGES_REQUESTED** [review]

File: changelog/rework_format_docs.dd (hunk 1 above)
Comment (berni44) [reply changelog/rework_format_docs.dd]: Dropped it. It's not that important. :-)

**Review by RazvanN7 — APPROVED** [review]
